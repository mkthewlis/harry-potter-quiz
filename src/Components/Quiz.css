.quiz__wrapper {
    display: flex;
    align-content: center;
    justify-content: center;
    flex-direction: column;
    color: #FFC500;
    height: 100vh;
    width: 100vw;
    background-color: #7F0909;
}

.quiz__header {
    display: flex;
    justify-content: center;
    text-transform: uppercase;
    font-size: 4em;
}

.quiz__question-title {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 2em;
}

.quiz__question {
    font-size: 1.5em;
}

.quiz__answer-container {
    display: flex;
    flex-direction: column;
    min-height: 20vh;
    justify-content: space-between;
    align-self: center;
}

button {
    width: 20vh;
    display: flex;
    align-self: center;
    justify-content: center;
}