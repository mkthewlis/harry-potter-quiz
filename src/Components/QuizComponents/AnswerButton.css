.answer-button {
    min-height: 1.8rem;
    min-width: 40rem;
    background-color: #FFFFFF;
    border-radius: 10px;
    align-items: center;
    cursor: pointer;
    border: 2px solid #FFC500;
    color: #7F0909;
    font-size: 1.5em;
    padding: .5rem;
    margin: 1rem 0;
    font-family: 'Courier New', Courier, monospace;
}