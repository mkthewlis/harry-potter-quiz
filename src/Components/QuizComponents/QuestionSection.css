.question-section__statistics-container {
    display: flex;
    justify-content: center;
    align-items: center;
}

.question-section__statistics-box {
    min-width: 5rem;
    min-height: 1.5rem;
    background-color: #FFC500;
    padding: 0.5rem;
    color: #7F0909;
    border: 2px solid #591313;
    border-radius: 5px;
    font-size: 1.3em;
    justify-content: center;
    margin-right: 5px;
}

.question-section__question-number {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 3em;
}

.question-section__question {
    font-size: 2em;
}

.question-section__answer-container {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-self: center;
}

.question-section__answer-container button {
    display: flex;
    align-self: center;
    justify-content: center;
}