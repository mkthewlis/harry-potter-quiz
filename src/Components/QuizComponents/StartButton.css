.start-button {
    min-height: 1.8rem;
    min-width: 20rem;
    background-color: #FFC500;
    border-radius: 10px;
    align-items: center;
    cursor: pointer;
    border: 2px solid #000000;
    color: #7F0909;
    font-size: 1.5em;
    padding: .5rem;
    margin-top: 4rem;
    text-transform: uppercase;
    font-family: 'Courier New', Courier, monospace;
}