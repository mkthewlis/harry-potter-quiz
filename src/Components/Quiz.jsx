import React, { useState } from 'react';
import { quizQuestions } from '../QuizData/QuizQuestions';
import { randomiseQuizQuestions } from '../Calculations/RandomiseQuizQuestions';
import { Button } from './QuizComponents/Button';
import './Quiz.css';

export const Quiz = () => {
    const tenRandomQuestions = randomiseQuizQuestions(quizQuestions);
    const [questionNumber, setQuestionNumber] = useState(0);
    const [result, setResult] = useState(0);

    const handleAnswerClick = (correctAnswer) => {
        if (correctAnswer === true && questionNumber < 9){
            setResult(result + 1);
        };
        const continueQuiz = questionNumber + 1;
        if (continueQuiz < 10){
            setQuestionNumber(continueQuiz);
        };
    };

    return(
    <div className="quiz__wrapper">
        <header>
            <h1 className="quiz__header">Harry Potter Pub Quiz</h1>
        </header>
        <main>
            <h2 className="quiz__question-number">Question {questionNumber + 1}/{tenRandomQuestions.length}</h2>
            <div className="quiz__statistics-container">
                <div className="quiz__statistics-box">Your score: {result}</div>
                <div className="quiz__statistics-box">Time remaining: 15.00</div>
            </div>
            <p className="quiz__question">{tenRandomQuestions[questionNumber].question}</p>
            <div className="quiz__answer-container">
                {tenRandomQuestions[questionNumber].possibleAnswers.map((possibleAnswer, index) => (
                    <Button onClick={() => handleAnswerClick(possibleAnswer.correctAnswer)} answer={possibleAnswer.answer}/>
                ))}
            </div>
        </main>
    </div>
    );
};
