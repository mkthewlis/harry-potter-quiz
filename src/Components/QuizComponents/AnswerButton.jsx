import React from 'react';
import './AnswerButton.css';

export const AnswerButton = ({index, answer, ...onClick}) => (
    <>
        <button key={index} className="answer-button" {...onClick}>{answer}</button>
    </>
);