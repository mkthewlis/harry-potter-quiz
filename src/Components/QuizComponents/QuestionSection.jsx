import React from 'react';
import { AnswerButton } from './AnswerButton';
import './QuestionSection.css';

export const QuestionSection = ({questionNumber, randomQuestions, countdownTimer, result, handleAnswerButtonClick}) => (
    <section>
        <h2 className="question-section__question-number">Question {questionNumber + 1}/{randomQuestions.length}</h2>
        <div className="question-section__statistics-container">
            <div className="question-section__statistics-box">Your score: {result}</div>
            <div className="question-section__statistics-box">{countdownTimer}s</div>
        </div>
        <p className="question-section__question">{randomQuestions[questionNumber].question}</p>
        <div className="question-section__answer-container">
            {randomQuestions[questionNumber].possibleAnswers.map((possibleAnswer) => (
                <AnswerButton onClick={() => handleAnswerButtonClick(possibleAnswer.correctAnswer)} answer={possibleAnswer.answer}/>
            ))}
        </div>
    </section>
);