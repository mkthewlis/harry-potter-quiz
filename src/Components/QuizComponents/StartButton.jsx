import React from 'react';
import './StartButton.css'

export const StartButton = ({...onClick}) => (
    <>
        <button className="start-button" {...onClick}>Start Quiz</button>
    </>
);