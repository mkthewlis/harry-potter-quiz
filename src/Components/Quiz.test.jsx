import React from 'react';
import { mount } from 'enzyme';
import { Quiz } from './Quiz';
import { StartButton } from './QuizComponents/StartButton';
import { QuestionSection } from './QuizComponents/QuestionSection';

describe('Quiz', () => {
    let wrapper;
    const onClickMock = jest.fn();
    const randomQuestionsMock = [
        {
          question: 'What do the initials J.K. Rowling stand for?',
          possibleAnswers: [
            { answer: 'Joanne Kathleen', correctAnswer: true },
            { answer: 'Jane Kathleen', correctAnswer: false },
            { answer: 'Jody Katie', correctAnswer: false },
            { answer: 'Joanne Katie', correctAnswer: false }
          ],
        }
    ];
    
    const questionNumberMock = 0;
    const countdownTimerMock = 15;
    const resultMock = 0;
    const handleAnswerButtonClick = jest.fn(); 
    beforeEach(() => {
        wrapper = mount(
          <Quiz randomQuestions={randomQuestionsMock}/>,
        );
      });
    it('renders without crashing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not render the results title by default', () => {
        expect(wrapper.find('.quiz__header').text()).toContain('Harry Potter Quiz');
    });

    it('should not call the start quiz button by default', () => {
        expect(onClickMock).not.toHaveBeenCalled();
    });
    describe('after the start button has been clicked', () => {
        it('should render the QuizSection component', () => {
            wrapper.find(StartButton).simulate('click');
            expect(wrapper.find('.quiz__header').exists()).toBeTruthy();
        });
    });
});     
