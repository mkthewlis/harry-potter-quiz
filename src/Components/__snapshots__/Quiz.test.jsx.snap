// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz renders without crashing 1`] = `
<Quiz
  randomQuestions={
    Array [
      Object {
        "possibleAnswers": Array [
          Object {
            "answer": "Joanne Kathleen",
            "correctAnswer": true,
          },
          Object {
            "answer": "Jane Kathleen",
            "correctAnswer": false,
          },
          Object {
            "answer": "Jody Katie",
            "correctAnswer": false,
          },
          Object {
            "answer": "Joanne Katie",
            "correctAnswer": false,
          },
        ],
        "question": "What do the initials J.K. Rowling stand for?",
      },
    ]
  }
>
  <div
    className="quiz__wrapper"
  >
    <header>
      <h1
        className="quiz__header"
      >
        Harry Potter Quiz
      </h1>
    </header>
    <main>
      <StartButton
        onClick={[Function]}
      >
        <button
          className="start-button"
          onClick={[Function]}
        >
          Start Quiz
        </button>
      </StartButton>
    </main>
  </div>
</Quiz>
`;
